### PR TITLE
[gainput] imporve cmake search gainput library

### PR DESCRIPTION
**Pull request by L-Sun**
**Describe the pull request**

- What does your PR fix? Fixes issue #10982 
Add a patch to improve CMake import [gainput]

### CODE REVIEW

File: ports/gainput/install_as_cmake_package.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     RUNTIME DESTINATION "bin"
+ )
++
++include(CMakePackageConfigHelpers)
++write_basic_package_version_file(
++    "${PROJECT_NAME}ConfigVersion.cmake"
++    VERSION ${GAINPUT_VERSION}
++    COMPATIBILITY SameMajorVersion
++)
++file(WRITE "${PROJECT_NAME}Config.cmake.in"
```
Comment (JackBoosY) [inline ports/gainput/install_as_cmake_package.patch:37]: Since this is not officially supported, please add `unofficial-` in front.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/gainput/install_as_cmake_package.patch (hunk 1 above)
Comment (L-Sun) [reply ports/gainput/install_as_cmake_package.patch:37]:
```suggestion
+file(WRITE "unofficial- ${PROJECT_NAME}Config.cmake.in" 
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
`gainput*.cmake` is still generated in _share/gainput_. We should generate `unofficial-gainput*.cmake` in _share/unofficial-gainput_.
And set `vcpkg_fixup_cmake_targets()` to `vcpkg_fixup_cmake_targets(CONFIG_PATH unofficial-gainput TARGET_PATH unofficial-gainput)`.
See [example](https://github.com/microsoft/vcpkg/blob/master/ports/box2d/portfile.cmake).

**Comment by L-Sun** [comment]
Thanks for giving a examples. I have modified the relevant code.Thanks for giving a examples. I have modified the relevant code.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Dynamic build generated `gainputstatic.lib`. Can you confirm this?

**Comment by L-Sun** [comment]
This behavior existed before I modified the code. Should I modify it?

**Comment by JackBoosY** [comment]
@L-Sun Yes, dynamic libraries can only be generated in dynamic triples, as are static libraries.

**Comment by L-Sun** [comment]
It has passed on x64_windows_static pipeline, so should we need update ci.baseline.txt with `gainput:x64-windows-static=pass`?

**Comment by JackBoosY** [comment]
@L-Sun Just remove `gainput:x64-windows-static=fail` from `ci.baseline.txt`.

**Comment by JackBoosY** [comment]
```
CMake Error at lib/CMakeLists.txt:58 (target_link_libraries):
  Cannot specify link libraries for target "gainput" which is not built by
  this project.
```

**Comment by L-Sun** [comment]
I don’t have an Apple device, so I can’t test in an osx environment. I have reset osx related code in
lib/CMakeLists.

**Comment by JackBoosY** [comment]
In _SOURCE/lib/CMakeLists.txt_ line 54:
```
elseif(APPLE)
...
        target_link_libraries(gainput ${FOUNDATION} ${IOKIT} ${GAME_CONTROLLER})
```

When using the triple `x64-osx` or `x64-linux`, the default library linkage is static, so it can only generate the target "gainputstatic", causing this error.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
No, the baseline seems incorrect and not related with this PR.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM, thanks for this PR!

**Comment by strega-nil** [comment]
Alright, this is cool! Thanks @L-Sun :)

**Pull request merged**